src/evaluators/MemberExpression.js

```diff
@@ -56,6 +59,25 @@ export default function(
   // 7. If the code matched by the syntactic production that is being evaluated is strict mode code, let strict be true, else let strict be false.
   let strict = strictCode;
 
-  // 8. Return a value of type Reference whose base value is bv, whose referenced name is propertyKey, and whose strict reference flag is strict.
-  return new Reference(bv, propertyKey, strict);
+  try {
+    // 8. Return a value of type Reference whose base value is bv, whose referenced name is propertyKey, and whose strict reference flag is strict.
+    return new Reference(bv, propertyKey, strict);
```

Comment:
Rather than relying on the constructor of Reference to do the checks and issue the diagnostic, do it here.